### PR TITLE
Ros2 api reorganization

### PR DESCRIPTION
**Pull request by pawelirh**
### Description

This PR aims to enhance the ROS 2 API naming and make it consistent across the project 

### Modifications

- Improve ROS2 API names
- Fix battery tests

## Summary by CodeRabbit

- **New Features**
  - Renamed `battery_node` to `battery_driver` for better clarity and consistency.
  - Updated topics for publishing and subscribing to enhance message organization.
  - Added configuration parameters for ADC device paths and moving average window lengths.

- **Bug Fixes**
  - Corrected indentation in `.github/workflows/release-repository.yaml`.

- **Documentation**
  - Updated README files across multiple packages to reflect new node names and configurations.

- **Refactor**
  - Renamed various nodes and topics for improved clarity and consistency.

- **Style**
  - Adjusted topic and service names for better readability and organization.

### CODE REVIEW

File: panther_localization/launch/localization.launch.py [hunk 1]
```diff
@@ -78,21 +78,21 @@ def generate_launch_description():
...
         parameters=[localization_config_path, {"tf_prefix": namespace}],
         namespace=namespace,
         remappings=[
             ("/diagnostics", "diagnostics"),
-            ("enable", "~/enable"),
-            ("set_pose", "~/set_pose"),
-            ("toggle", "~/toggle"),
+            ("enable", "localization/enable"),
+            ("set_pose", "localization/set_pose"),
+            ("toggle", "localization/toggle"),
```
[the remark this answers is not in the document]
Comment (pawelirh) [reply panther_localization/launch/localization.launch.py:91]: Everything is fine, the executable name is indeed `ekf_node` and in line 90 remapping rule is defined so we need to refer to the old name.

File: panther_bringup/README.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ The package contains the default configuration and launch files necessary to sta
...
 - `wheel_type` [*string*, default: **WH01**]: type of wheel, possible are: **WH01** - off-road, **WH02** - mecanum, **WH04** - small pneumatic, and **custom** - custom wheel types (requires setting `wheel_config_path` argument accordingly).
-- `components_config_path` [*string*, default: **None**]: Additional components configuration file. Components described in this file          are dynamically included in Panther's urdf. Panther options are described in [the manual](https://husarion.com/manuals/panther/panther-options. Example of configuration you can find [config/components.yaml](config/components.yaml)
+- `components_config_path` [*string*, default: **None**]: Additional components configuration file. Components described in this file          are dynamically included in Panther's urdf. Panther options are described in [the manual](<https://husarion.com/manuals/panther/panther-options>. Example of configuration you can find [config/components.yaml](config/components.yaml)
```
Comment (KmakD) [inline panther_bringup/README.md]:
```suggestion
- `components_config_path` [*string*, default: **None**]: Additional components configuration file. Components described in this file          are dynamically included in Panther's urdf. Panther options are described in [the manual](https://husarion.com/manuals/panther/panther-options). Example of configuration you can find [config/components.yaml](config/components.yaml)
```
Isn't this simply missing a bracket?

File: panther_controller/config/WH01_controller.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
       joint_state_broadcaster:
         type: joint_state_broadcaster/JointStateBroadcaster
-      panther_base_controller:
+      drive_controller:
```
Comment (KmakD) [inline panther_controller/config/WH01_controller.yaml:8]: Have you considered `base_controller`? I feel like it is more commonly used.

File: panther_localization/README.md [hunk 4]
```diff
@@ -20,27 +20,27 @@ This package contains:
...
 - `navsat_transform` [*[robot_localization/navsat_transform_node](https://github.com/cra-ros-pkg/robot_localization/tree/ros2)*]:
 It converts raw GPS data into odometry data and publishes corrected GPS positions based on sensor data at a higher frequency.
 
-#### ekf_node - Subscriber
+#### ekf_filter - Subscriber
 
 - `cmd_vel` [*geometry_msgs/msg/Twist*]: command velocity value.
 - `odometry/wheels` [*nav_msgs/msg/Odometry*]: robot odometry calculated from wheels.
 - `imu/data` [*sensor_msgs/msg/Imu*]: filtered IMU data.
+- `localization/set_pose` [*geometry_msgs/msg/PoseWithCovarianceStamped*]: state of the filter.
```
Comment (KmakD) [inline panther_localization/README.md]: This allows to set robot's position similar to a service

**Review by KmakD — CHANGES_REQUESTED** [review]

File: panther_controller/config/WH01_controller.yaml (hunk 3 above)
Comment (pawelirh) [reply panther_controller/config/WH01_controller.yaml:8]: Yep, but having both `base_controller` and `hardware_controller` seems to me a bit confusing what's for what. The `drive_controller` explicitly highlights what is behind this node.

File: panther_localization/README.md (hunk 4 above)
Comment (pawelirh) [reply panther_localization/README.md]: Yes, that's right. 🤔  Do you find the description imprecise?

Comment (KmakD) [reply panther_localization/README.md]: A bit, I'm missing information that this topic allows to change state of the filter.

Comment (pawelirh) [reply panther_localization/README.md]: The description is elaborated now 🏁

**Review by KmakD — APPROVED** [review]

**Pull request merged**